parametric/test_headers_b3.py

```diff
@@ -0,0 +1,131 @@
+from typing import Any
+
+import pytest
+
+from parametric.spec.trace import SAMPLING_PRIORITY_KEY, ORIGIN
+from parametric.spec.trace import span_has_no_parent
+from parametric.utils.headers import make_single_request_and_get_inject_headers
+from parametric.utils.test_agent import get_span
+
+parametrize = pytest.mark.parametrize
+
+
+def enable_b3() -> Any:
+    env1 = {
+        "DD_TRACE_PROPAGATION_STYLE_EXTRACT": "B3 SINGLE HEADER",
+        "DD_TRACE_PROPAGATION_STYLE_INJECT": "b3 single header",
+    }
+    env2 = {
+        "DD_TRACE_PROPAGATION_STYLE": "B3 single header",
+    }
+    return parametrize("library_env", [env1, env2])
+
+
+@enable_b3()
+@pytest.mark.skip_library("golang", "not implemented")
+@pytest.mark.skip_library("nodejs", "TODO: remove when https://github.com/DataDog/dd-trace-js/pull/2477 lands")
+def test_headers_b3_extract_valid(test_agent, test_library):
+    """Ensure that b3 distributed tracing headers are extracted
+    and activated properly.
+    """
+    with test_library:
+        headers = make_single_request_and_get_inject_headers(
+            test_library, [["b3", "000000000000000000000000075bcd15-000000003ade68b1-1"]]
+        )
+
+    span = get_span(test_agent)
+    assert span.get("trace_id") == 123456789
+    assert span.get("parent_id") == 987654321
+    assert span["metrics"].get(SAMPLING_PRIORITY_KEY) == 1
+    assert span["meta"].get(ORIGIN) is None
+
+
+@enable_b3()
+@pytest.mark.skip_library("golang", "not implemented")
+@pytest.mark.skip_library("nodejs", "TODO: remove when https://github.com/DataDog/dd-trace-js/pull/2477 lands")
+def test_headers_b3_extract_invalid(test_agent, test_library):
+    """Ensure that invalid b3 distributed tracing headers are not extracted.
+    """
+    with test_library:
+        headers = make_single_request_and_get_inject_headers(test_library, [["b3", "0-0-1"]])
+
+    span = get_span(test_agent)
+    assert span.get("trace_id") != 0
+    assert span_has_no_parent(span)
+    assert span["meta"].get(ORIGIN) is None
+
+
+@enable_b3()
+@pytest.mark.skip_library("python", "Issue: Python doesn't pad the trace-id to length of 16 or 32 lower-hex characters")
```

Comment:
@ZStriker19 Is this still an issue

Reply:
I added this skip very recently when because I noticed this test failure, so I think it is still an issue that will need to be addressed

Reply:
I'll make the fix.

Reply:
https://github.com/DataDog/dd-trace-py/pull/4976

Reply:
We can unskip the test in a separate PR later so that we can merge this large set of changes right away